plugins/aws/resoto_plugin_aws/resource/redshift.py

```diff
@@ -0,0 +1,445 @@
+        "estimated_time_to_completion_in_seconds": S("EstimatedTimeToCompletionInSeconds"),
+        "elapsed_time_in_seconds": S("ElapsedTimeInSeconds"),
+    }
+    status: Optional[str] = field(default=None)
+    current_rate_in_mega_bytes_per_second: Optional[float] = field(default=None)
+    total_data_in_mega_bytes: Optional[int] = field(default=None)
+    data_transferred_in_mega_bytes: Optional[int] = field(default=None)
+    estimated_time_to_completion_in_seconds: Optional[int] = field(default=None)
+    elapsed_time_in_seconds: Optional[int] = field(default=None)
+
+
+@define(eq=False, slots=False)
+class AwsRedshiftHsmStatus:
+    kind: ClassVar[str] = "aws_redshift_hsm_status"
+    mapping: ClassVar[Dict[str, Bender]] = {
+        "hsm_client_certificate_identifier": S("HsmClientCertificateIdentifier"),
+        "hsm_configuration_identifier": S("HsmConfigurationIdentifier"),
+        "status": S("Status"),
+    }
+    hsm_client_certificate_identifier: Optional[str] = field(default=None)
+    hsm_configuration_identifier: Optional[str] = field(default=None)
+    status: Optional[str] = field(default=None)
+
+
+@define(eq=False, slots=False)
+class AwsRedshiftClusterSnapshotCopyStatus:
+    kind: ClassVar[str] = "aws_redshift_cluster_snapshot_copy_status"
+    mapping: ClassVar[Dict[str, Bender]] = {
+        "destination_region": S("DestinationRegion"),
+        "retention_period": S("RetentionPeriod"),
+        "manual_snapshot_retention_period": S("ManualSnapshotRetentionPeriod"),
+        "snapshot_copy_grant_name": S("SnapshotCopyGrantName"),
+    }
+    destination_region: Optional[str] = field(default=None)
+    retention_period: Optional[int] = field(default=None)
+    manual_snapshot_retention_period: Optional[int] = field(default=None)
+    snapshot_copy_grant_name: Optional[str] = field(default=None)
+
+
+@define(eq=False, slots=False)
+class AwsRedshiftClusterNode:
+    kind: ClassVar[str] = "aws_redshift_cluster_node"
+    mapping: ClassVar[Dict[str, Bender]] = {
+        "node_role": S("NodeRole"),
+        "private_ip_address": S("PrivateIPAddress"),
+        "public_ip_address": S("PublicIPAddress"),
+    }
+    node_role: Optional[str] = field(default=None)
+    private_ip_address: Optional[str] = field(default=None)
+    public_ip_address: Optional[str] = field(default=None)
+
+
+@define(eq=False, slots=False)
+class AwsRedshiftElasticIpStatus:
+    kind: ClassVar[str] = "aws_redshift_elastic_ip_status"
+    mapping: ClassVar[Dict[str, Bender]] = {"elastic_ip": S("ElasticIp"), "status": S("Status")}
+    elastic_ip: Optional[str] = field(default=None)
+    status: Optional[str] = field(default=None)
+
+
+@define(eq=False, slots=False)
+class AwsRedshiftClusterIamRole:
+    kind: ClassVar[str] = "aws_redshift_cluster_iam_role"
+    mapping: ClassVar[Dict[str, Bender]] = {"iam_role_arn": S("IamRoleArn"), "apply_status": S("ApplyStatus")}
+    iam_role_arn: Optional[str] = field(default=None)
```

Comment:
should this result in a `dependant_node` to the corresponding instance of `AwsIamRole`?